### PR TITLE
Make function get_progress be compatible with more video format

### PR DESCRIPTION
**Pull request by Thatwho**
* Add function _get_duration with VideoReader
* Update function get_progress of VideoReader

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by azhavoro** [comment]
@Thatwho Thanks for the your contribution!

**Comment by nmanovic** [comment]
@Thatwho , should we mention an issue which the PR resolves? Could you please add a line into CHANGELOG.md?

File: CHANGELOG.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Duplication of the cuboids when redraw them (<https://github.com/openvinotoolkit/cvat/pull/3308>)
 - Some code issues in Deep Extreme Cut handler code (<https://github.com/openvinotoolkit/cvat/pull/3325>)
 - UI fails when inactive user is assigneed to a task/job (<https://github.com/openvinotoolkit/cvat/pull/3343>)
+- Can't calculat precise progress of image extraction and compression when operating video with particular format (https://github.com/openvinotoolkit/cvat/pull/3381)
```
Comment (nmanovic) [inline CHANGELOG.md:45]:
```suggestion
- Calculate precise progress of decoding a video file (https://github.com/openvinotoolkit/cvat/pull/3381)
```

**Review by nmanovic — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -42,6 +42,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Duplication of the cuboids when redraw them (<https://github.com/openvinotoolkit/cvat/pull/3308>)
 - Some code issues in Deep Extreme Cut handler code (<https://github.com/openvinotoolkit/cvat/pull/3325>)
 - UI fails when inactive user is assigneed to a task/job (<https://github.com/openvinotoolkit/cvat/pull/3343>)
+- Calculate precise progress of decoding a video file (https://github.com/openvinotoolkit/cvat/pull/3381)
```
Comment (nmanovic) [inline CHANGELOG.md:45]:
```suggestion
- Calculate precise progress of decoding a video file (<https://github.com/openvinotoolkit/cvat/pull/3381>)
```

**Comment by nmanovic** [comment]
@Thatwho , thanks for the contribution!

**Pull request merged**